public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AzureCredentialsForm.tsx

```diff
@@ -72,7 +94,7 @@ export const AzureCredentialsForm: FunctionComponent<Props> = (props: Props) =>
   };
 
   const onTenantIdChange = (event: ChangeEvent<HTMLInputElement>) => {
-    if (onCredentialsChange) {
+    if (onCredentialsChange && credentials.authType === 'clientsecret') {
```

Comment:
The callback won't be called for anything over than `clientsecret` because otherwise controls are hidden but this check needed for type safety, to cast to the type. `credentials` is of generic type, so we should cast to `clientsecret` to be able to access the fields